### PR TITLE
Upgrade ClickHouse version to 23.8-lts

### PR DESCRIPTION
**Pull request by orginux**
Signed-off-by: Denis Gorobets [orginux@protonmail.com](mailto:orginux@protonmail.com)

## Important items to consider before making a Pull Request
Please check items PR complies to:
* [x] All commits in the PR are squashed. [More info](https://github.com/Altinity/clickhouse-operator/blob/master/CONTRIBUTING.md#how-to-make-pr) 
* [x] The PR is made into dedicated `next-release` branch, **not into** `master` branch<sup>1</sup>. [More info](https://github.com/Altinity/clickhouse-operator/blob/master/CONTRIBUTING.md#how-to-make-pr)
* [x] The PR is signed. [More info](https://github.com/Altinity/clickhouse-operator/blob/master/CONTRIBUTING.md#sign-your-work)

--

<sup>1</sup> If you feel your PR does not affect any Go-code or any testable functionality (for example, PR contains docs only or supplementary materials), PR can be made into `master` branch, but it has to be confirmed by project's maintainer.

### CODE REVIEW

**Comment by Slach** [comment]
just search and replace version in yaml files is not enough, it could break something like test which use

some files shall to be excluded

File: tests/e2e/manifests/chi/test-017-multi-version.yaml [hunk 1]
```diff
@@ -9,12 +9,12 @@ spec:
       spec:
         containers:
         - name: clickhouse-pod
-          image: clickhouse/clickhouse-server:22.3
+          image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chi/test-017-multi-version.yaml:12]: 23.3 shall be here

File: tests/e2e/manifests/chi/test-006-ch-upgrade-3.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ spec:
       spec:
         containers:
         - name: clickhouse-pod
-          image: clickhouse/clickhouse-server:22.3
+          image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chi/test-006-ch-upgrade-3.yaml:12]: 23.3 shall be here

File: tests/e2e/manifests/chi/test-006-ch-upgrade-1.yaml [hunk 3]
```diff
@@ -9,7 +9,7 @@ spec:
       spec:
         containers:
         - name: clickhouse-pod
-          image: clickhouse/clickhouse-server:22.3
+          image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chi/test-006-ch-upgrade-1.yaml:12]: 23.3 here

File: docs/chi-examples/evolution/04-update-introduce-canary.yaml [hunk 4]
```diff
@@ -35,7 +35,7 @@ spec:
         spec:
           containers:
             - name: clickhouse
-              image: clickhouse/clickhouse-server:21.8
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/evolution/04-update-introduce-canary.yaml:38]: 23.3 here

File: docs/chi-examples/06-advanced-layout-04-multiple-clusters.yaml [hunk 5]
```diff
@@ -48,16 +48,16 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.1
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/06-advanced-layout-04-multiple-clusters.yaml:51]: 22.8 here

File: docs/chi-examples/06-advanced-layout-04-multiple-clusters.yaml [hunk 6]
```diff
@@ -48,16 +48,16 @@ spec:
...
 
       - name: t2
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.2
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/06-advanced-layout-04-multiple-clusters.yaml:57]: 23.3 here

File: docs/chi-examples/06-advanced-layout-03-replicas.yaml [hunk 7]
```diff
@@ -32,17 +32,17 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.1
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/06-advanced-layout-03-replicas.yaml:35]: 22.8 here

File: docs/chi-examples/06-advanced-layout-03-replicas.yaml [hunk 8]
```diff
@@ -32,17 +32,17 @@ spec:
...
 
       - name: clickhouse:22.3
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.2
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/06-advanced-layout-03-replicas.yaml:41]: 23.3

File: docs/chi-examples/06-advanced-layout-02-shards.yaml [hunk 9]
```diff
@@ -32,17 +32,17 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.1
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/06-advanced-layout-02-shards.yaml:35]: 22.8

File: docs/chi-examples/06-advanced-layout-02-shards.yaml [hunk 10]
```diff
@@ -32,17 +32,17 @@ spec:
...
 
       - name: clickhouse:22.2
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.2
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/06-advanced-layout-02-shards.yaml:41]: 23.3

File: tests/e2e/manifests/chit/tpl-clickhouse-21.3.yaml [hunk 11]
```diff
@@ -13,4 +13,4 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:21.3
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chit/tpl-clickhouse-21.3.yaml:16]: rollback it

File: tests/e2e/manifests/chit/tpl-clickhouse-21.8.yaml [hunk 12]
```diff
@@ -13,4 +13,4 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:21.8
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chit/tpl-clickhouse-21.8.yaml:16]: rollback it

File: tests/e2e/manifests/chit/tpl-clickhouse-22.1.yaml [hunk 13]
```diff
@@ -13,4 +13,4 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.1
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chit/tpl-clickhouse-22.1.yaml:16]: rollback it

File: tests/e2e/manifests/chit/tpl-clickhouse-22.2.yaml [hunk 14]
```diff
@@ -13,4 +13,4 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.2
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chit/tpl-clickhouse-22.2.yaml:16]: rollback it

File: tests/e2e/manifests/chit/tpl-clickhouse-22.3.yaml [hunk 15]
```diff
@@ -13,4 +13,4 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.3
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chit/tpl-clickhouse-22.3.yaml:16]: rollback it

File: tests/e2e/manifests/chit/tpl-clickhouse-22.6.yaml [hunk 16]
```diff
@@ -13,5 +13,5 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.6
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chit/tpl-clickhouse-22.6.yaml:16]: rollback it

File: tests/e2e/manifests/chit/tpl-clickhouse-22.7.yaml [hunk 17]
```diff
@@ -13,5 +13,5 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.7
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chit/tpl-clickhouse-22.7.yaml:16]: rollback it

File: tests/e2e/manifests/chit/tpl-clickhouse-22.8.yaml [hunk 18]
```diff
@@ -13,5 +13,5 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.8
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline tests/e2e/manifests/chit/tpl-clickhouse-22.8.yaml:16]: rollback it

Comment (Slach) [reply tests/e2e/manifests/chit/tpl-clickhouse-22.8.yaml:16]: add separate `tpl-clickhouse-23.8.yaml`

and look references to tpl in `tests`

**Review by Slach — CHANGES_REQUESTED** [review]
please rollback changes in some files and use properly versions in some other

File: tests/e2e/manifests/chit/tpl-clickhouse-22.8.yaml (hunk 18 above)
Comment (orginux) [reply tests/e2e/manifests/chit/tpl-clickhouse-22.8.yaml:16]: As far as I can see the file is already present: https://github.com/Altinity/clickhouse-operator/blob/master/tests/e2e/manifests/chit/tpl-clickhouse-23.8.yaml

File: docs/chi-examples/06-advanced-layout-01-one-shard-many-replicas.yaml [hunk 19]
```diff
@@ -21,10 +21,10 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.1
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/06-advanced-layout-01-one-shard-many-replicas.yaml:24]: sync template name with and ferenece to template name in line 13 and line 20 above

File: docs/chi-examples/06-advanced-layout-02-shards.yaml [hunk 20]
```diff
@@ -32,17 +32,17 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.1
+              image: clickhouse/clickhouse-server:22.8
 
       - name: clickhouse:22.2
```
[the remark this answers is not in the document]
Comment (Slach) [reply docs/chi-examples/06-advanced-layout-02-shards.yaml:37]: sync names in line 23, 18, 13 above

File: docs/chi-examples/08-clickhouse-version-update-02-apply-update-one.yaml [hunk 21]
```diff
@@ -23,10 +23,10 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.2
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/08-clickhouse-version-update-02-apply-update-one.yaml:26]: sync podTemplate name nad reference in line 22, 18,10, 28

File: docs/chi-examples/99-clickhouseinstallation-max.yaml [hunk 22]
```diff
@@ -569,7 +569,7 @@ spec:
         spec:
           containers:
             - name: clickhouse
-              image: clickhouse/clickhouse-server:22.3
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/99-clickhouseinstallation-max.yaml:572]: look above to template name line 489, sync it

File: docs/chi-examples/99-clickhouseinstallation-max.yaml [hunk 23]
```diff
@@ -599,7 +599,7 @@ spec:
         spec:
           containers:
             - name: clickhouse
-              image: clickhouse/clickhouse-server:22.2
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/chi-examples/99-clickhouseinstallation-max.yaml:602]: change template name in line 593 and find usages of this template

File: docs/custom_resource_explained.md [hunk 24]
```diff
@@ -561,7 +561,7 @@ with additional sections, such as:
         spec:
           containers:
             - name: clickhouse
-              image: clickhouse/clickhouse-server:22.3
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/custom_resource_explained.md:564]: sync description in line 543 and 542 with actual clickhouse version

File: docs/replication_setup.md [hunk 25]
```diff
@@ -48,7 +48,7 @@ spec:
         spec:
           containers:
             - name: clickhouse-pod
-              image: clickhouse/clickhouse-server:22.3
+              image: clickhouse/clickhouse-server:23.8
```
Comment (Slach) [inline docs/replication_setup.md:51]: sync with name in line 47 and find usages this template and change template name reference too

File: tests/e2e/manifests/chi/test-022-broken-image.yaml [hunk 26]
```diff
@@ -12,7 +12,7 @@ spec:
       spec:
         containers:
         - name: clickhouse-pod
-          image: clickhouse/clickhouse-server:22.8-broken
+          image: clickhouse/clickhouse-server:23.8-broken
```
Comment (Slach) [inline tests/e2e/manifests/chi/test-022-broken-image.yaml:15]: sync with line 11 and 8

**Review by Slach — CHANGES_REQUESTED** [review]
please sync template images with templates names and link to these template names in other parts of manifests

**Pull request merged**